datastream-server/src/main/java/com/linkedin/datastream/server/assignment/LoadBasedPartitionAssigner.java

```diff
@@ -144,6 +169,14 @@
       newAssignments.put(instance, newTasks);
     });
 
+    // update metrics
+    PartitionAssignmentStats stats = new PartitionAssignmentStats(minPartitionsAcrossTasks.get(),
```

Comment:
I think it makes the code cleaner. If I remove it, I'll have to maintain two separate maps for min/max partitions across tasks per datastream. This is more extensible as we can add more stats and add more per-datastream metrics in the future (e.g. the maxThroughputAcrossTasks we discussed earlier).

Reply:
The major problem I can see with this future extension is, if there is another method created and that method calculates some stats, then either you will have to define all the getter and setters and make the fields non-final. Suppose we want to create any alert based metric, then all the other fields also need to be set. To overcome this, you may have to optional to figure out which fields are set and it will be an overkill.

You can have separate methods for each metric for create/Update the metric and just call those method, just like it is done in other classes in this repo.

Reply:
I'm not sure what you mean by an alert-based metric, but from the description it seems like it doesn't fit with what I see as partition assignment stats. Semantically they might be incompatible, and we don't have to put it in the class. I still see value in keeping `PartitionAssignmentStats` around, for the reasons I mentioned above.

Reply:
I mean if you have to emit a new metric which is in the exception path (and you don't calculate the other stats), then reusing this stat will be an issue and will require refactoring, V/s addition of new metric not impacting any other code change.